### PR TITLE
chore: repo cleaning, remove unused code paths

### LINKED ISSUE #81: Test Organization
Goal: Remove cognitive load by constraining tests to their own subdirectory so we don't mix the circomkit autogenerated files with the test files.

Also Remove the `inputs/` directory as we are not using it for testing. 

Bonus: Investigate circomkit configuration to see if we can do anything about this.

### LINKED ISSUE #82: Remove Unused Code Paths
We should remove all unused code paths. I think this involves removing much of the code in `AES-CTR/`, removing redundant code paths, and also remove and addressing stale TODOs as well as working towards idiomatic naming of templates and variables.

### PR DESCRIPTION
**Pull request by thor314**
Closes #82.
Closes #81 

- add circomlib in circuits. Fixes circom-lsp path issues.
- silence logs in tests
- refactor codebase to remove unused code paths (in progress)

Note: All tests pass right now except the aes-gcm one because the new components need to get plugged in correctly. @devloper I assume you grabbed those?

### CODE REVIEW

**Comment by thor314** [comment]
work started, a lot of circom to clean up and de-duplicate. 

There are several tests that fail to compile at the moment, most notably the aes-gcm circuit appears to be broken. @0xJepsen maybe we could fix this when we pair. repro: `npx mocha -g aes-gcm`

**Comment by devloper** [comment]
Why duplicate the circomlib code instead of importing?

**Comment by thor314** [comment]
what do you mean @devloper

**Comment by devloper** [comment]
This pull requests copies all of the circuits from circomlib into the repository.  With circom you can specify an include flag, such as `circom -l ./node_modules` to include the circomlib circuits from an import.  Circomkit itself does this by default. As a result there is no need to copy all the circuits.

**Comment by thor314** [comment]
removed circomlib, added to my gitignore until further discussion

**Comment by 0xJepsen** [comment]
@devloper What do you think about removing the `client` directory? Are we still using that?

File: circuits/test/aes/cipher.test.ts [hunk 1]
```diff
@@ -65,3 +64,36 @@ describe("Cipher", () => {
     );
   });
 });
+
```
Comment (thor314) [inline circuits/test/aes/cipher.test.ts:67]: what's the deal here?

File: circuits/test/gfmulx/ghash_mulx.test.ts
Comment (thor314) [inline circuits/test/gfmulx/ghash_mulx.test.ts:1]: i thought you used mulx in your implementation?

File: circuits/test/aes/cipher.test.ts (hunk 1 above)
Comment (0xJepsen) [reply circuits/test/aes/cipher.test.ts:67]: 💩

**Review by thor314 — COMMENTED** [review]
seems mostly renaming and code deletion, good stuff. I'm not sure what the inputs stuff was used for, pending tracy's approval

File: circuits/test/aes/cipher.test.ts (hunk 1 above)
Comment (thor314) [reply circuits/test/aes/cipher.test.ts:67]: fair

**Comment by 0xJepsen** [comment]
`-4000`

File: circuits/test/gfmulx/ghash_mulx.test.ts
Comment (0xJepsen) [reply circuits/test/gfmulx/ghash_mulx.test.ts:1]: I do but not the one this tested / i moved it into the mul_test

**Pull request merged**